### DISCUSSION TITLE
interactivity in javascript

### DISCUSSION THREAD

**Discussion by maximestephan** [molstar/molstar#689] [Q&A]
Hi ,

we try to get the selected atom indexes, but with no success.

we only get the number of atom selected with the following javascript :
BasicMolStarWrapper.plugin.managers.structure.hierarchy.current.structures[0].selection.elementCount

Any chance that there is a function like  BasicMolStarWrapper.plugin.managers.structure.hierarchy.current.structures[0].selection.getSelectedAtomIndex() that return a array with the index of the atom selected ? 

Regards,
Maxime

**Comment by dsehnal** [comment]
`plugin.managers.structure.selection.getLoci(...)` should help. I've added https://github.com/molstar/molstar/commit/2e013fafc835da774134d11879f82c04261fc85d to make the iteration over it simpler.

You can either use the `element` field of the loci (the index of the atom in the internal representation) or use `StructureProperties.atom.sourceIndex` to get the index of the atom in the source file.

**Comment by maximestephan** [comment]
thanks!
BasicMolStarWrapper.plugin.managers.structure.selection.getLoci() returns 'empty-locki'  

sometimes , the following code is working, but sometime not , depending on the selection.
BasicMolStarWrapper.plugin.managers.structure.selection.entries.forEach( function(e) {
    console.log(e.selection.elements[0].indices);
});
